testing/environments/docker/kafka/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM debian:buster
+FROM debian:12
```

Comment:
The rest of the repository is using release names instead of numbers, see https://github.com/search?q=repo%3Aelastic%2Fbeats%20%22debian%3A%22&type=code

I think it's better to stay consistent across the repository and either keep using names here or change the rest of the images to numbers too.